dali/base/dadfs.cpp

```diff
@@ -4093,9 +4093,9 @@ protected: friend class CDistributedFilePart;
 
     virtual IDistributedFilePart &queryPart(unsigned idx) override
     {
-        if (idx<parts.ordinality())
-            return queryParts().item(idx);
-        return *(IDistributedFilePart *)NULL;
+        if (idx>=parts.ordinality())
+            assertex(!"CDistributedFileBase::queryPart out of range");
```

Comment:
style: throwUnexpectedX("CDistributedFileBase::queryPart out of range") is cleaner/more modern.

Reply:
same in the function below

Reply:
yep, I'll change them.

Reply:
@ghalliday - changed and squashed in.